### PR TITLE
单元测试&logging

### PR DESCRIPTION
**Pull request by youqingxiaozhua**
- 完善了单元测试，token0全流程
- 修复了log error
- 错误信息记录成error便于查看

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -53,15 +54,9 @@ def send_email(to_email, subject, content, attachment=None):
                        smtp_skip_login=email_json["skip_auth"])
     # 邮箱正文
     # contents = ['您好，附件中有您的证书']
-    logger.info("send to " + to_email, 'with subject', subject)
+    logger.info("send to " + to_email + 'with subject' + subject)
```
Comment (zhaofeng-shu33) [inline utils.py]: This line is in PR #106

File: test.py [hunk 2]
```diff
@@ -105,7 +94,8 @@ def test_insert_people(self):
             insert_people('abc@example.org', 'fake name')
         except:
             pass
- 
+
```
Comment (zhaofeng-shu33) [inline test.py]: please remove extra empty lines.

File: test.py [hunk 3]
```diff
@@ -84,19 +86,6 @@ def test_submitSendEmailRequest(self):
         res_json = json.loads(response.data.decode('ascii'))
         self.assertEqual(res_json['code'], 1)
 
-class SubmitUserInfoTests(unittest.TestCase):
```
Comment (zhaofeng-shu33) [inline test.py]: Why remove this test?

File: test.py [hunk 4]
```diff
@@ -1,71 +1,76 @@
...
     def test_getSMTP_url(self):
         smtp_url = get_smtp_url('foo@163.com')
         self.assertEqual(smtp_url, 'smtp.163.com')
 
+
 class WebAPITests(unittest.TestCase):
-    def test_getUserInfo(self):
+    def test_0_send_email(self):
+        with patch("yagmail.SMTP") as mock_smtp:
+            SendEmailJob().send_notice_email()  # threading中raise的错误不会被catch
```
Comment (zhaofeng-shu33) [inline test.py]: test_1, test_2, test_3 are serialized executed?

File: utils.py (hunk 1 above)
Comment (youqingxiaozhua) [reply utils.py]: close duplicate PR or just let git deal with this

File: test.py (hunk 3 above)
Comment (youqingxiaozhua) [reply test.py]: Because it is tested in the API of `/API/submitUserInfo`

File: test.py (hunk 4 above)
Comment (youqingxiaozhua) [reply test.py]: yep

File: utils.py (hunk 1 above)
Comment (zhaofeng-shu33) [reply utils.py]: ok

**Pull request merged**